docs/spec/v1alpha1/common.md

```diff
@@ -0,0 +1,123 @@
+# Common
+
+Common defines resources used across all source types.
+
+## Specification
+
+### Source synchronization
+
+Source objects should contain a `spec.interval` field that tells the controller at which interval to check for updates:
+
+```go
+type SourceSpec struct {
```

Comment:
I am wondering if the requirement of the presence of an `Interval` field in combination with the expected presence of an `Artifact` field in the status of the resource gives reason to introduce a common source interface exposing both those fields.

Reply:
It would look something like:

```go
type Source interface {
  Interval() metav1.Duration
  Artifact() *Artifact
}

Reply:
Added in https://github.com/fluxcd/source-controller/pull/3/commits/f57d9ec770a8ef3159408d319bf3fe603e896d3c